### PR TITLE
Support for GNOME 45, drop support for previous GNOME versions

### LINKED ISSUE #362: Support for GNOME 45
We need to add [support for GNOME 45](https://gjs.guide/extensions/upgrading/gnome-shell-45.html).

I've pushed pushing my code with GNOME 45 support to the **[362-support-for-gnome-45 branch](https://github.com/projecthamster/hamster-shell-extension/tree/362-support-for-gnome-45)** branch in this repository.

The branch seems to work. I'm using it right now. Please test and please post any issues here.

### PR DESCRIPTION
**Pull request by mwilck**
This PR fixes #362. When it is merged, we will have to rework our sticky issue #327.

### CODE REVIEW

**Comment by jacek-pliszka** [comment]
OK, it does not fully work for me.

It does not show up on the screen.  

Only super-t works.

How to fix it?

File: extension/prefs.js [hunk 1]
```diff
@@ -21,156 +21,132 @@ Copyright (c) 2016 - 2018 Eric Goller / projecthamster <elbenfreund@projecthamst
...
+        return Gtk.accelerator_valid(key, mods) && Gtk.accelerator_name(key, mods);
+      });
+
+      if (mappings.every((x) => !!x)) {
+	console.log("HotkeyRow: good hotkey value ", hotkeys);
+	this.current = mappings;
+	settings.set_strv(bind, this.current);
+      } else {
+	console.log("invalid hotkey value ", hotkeys);
+	this.set_text(this.current.join(","));
```
Comment (hedayat) [inline extension/prefs.js:58]: Doesn't seem to be properly indented.

File: extension/prefs.js [hunk 2]
```diff
@@ -21,156 +21,132 @@ Copyright (c) 2016 - 2018 Eric Goller / projecthamster <elbenfreund@projecthamst
...
+    ];
+
+    for (const {p, title} of placements) {
+      const btn = new Gtk.CheckButton({
+	action_name:   'hamster.panel-placement',
+	action_target: new GLib.Variant('i', p),
+      });
+      const row = new Adw.ActionRow({
+	activatable_widget: btn,
+	title,
```
Comment (hedayat) [inline extension/prefs.js:105]: Indentation problem here too. and some other locations below, which I won't comment :P

**Review by hedayat — APPROVED** [review]
LGTM. I just noted some indentation problems. Other than that, looks fine. (Although, I'm not a JS coder anyway :P )

**Comment by 3-w-c** [comment]
Tested here, working great for me!

**Comment by piio** [comment]
Hi, after updating to 
[362-support-for-gnome-45](https://github.com/projecthamster/hamster-shell-extension/tree/362-support-for-gnome-45), I have following error:
Hamster: connection to DBUS window service failed
Connection to DBUS window service failed
TypeError: _actor.add_actor is not a function

**Comment by mwilck** [comment]
@piio

> Hamster: connection to DBUS window service failed

That sounds more like an issue with hamster itself. Did you perhaps update not only the GNOME extension, but also hamster?

Can you run "hamster" from the command line? Does the extension work after you do?

**Comment by piio** [comment]
The hamster processess are running - after running hamster from command line there is no difference.
I tried to use both master from github version and latest stable 3.0.3.

I will try to test on fresh user profile.

**Comment by c0m1c5an5** [comment]
Can confirm, everything works properly. Fedora39, Gnome 45.5, Hamster master branch (81be67e08a6c2a69f2f042df5459dba886cbd7de).

**Comment by mwilck** [comment]
As GNOME 45 is out for 6 months now, I've merged this into the `develop` branch.

**Pull request merged**